zetasql/parser/bison_parser.y

```diff
@@ -5534,6 +5542,10 @@ expression:
           binary_expression->set_op(zetasql::ASTBinaryExpression::LIKE);
           $$ = binary_expression;
         }
+    | expression "ESCAPE" string_literal
```

Comment:
yet, there is no restriction to use 'ESCAPE' with operator other than 'LIKE'. 
Upper phase will assert that